tests/slist_ut/slist_ut.c

```diff
@@ -0,0 +1,1114 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+#include "macro_utils/macro_utils.h"
+#include "umock_c/umock_c.h"
+#include "umock_c/umocktypes_bool.h"
+#include "umock_c/umocktypes_charptr.h"
+
+
+#include "c_util/slist.h"
+#include "c_util/containing_record.h"
+#include "testrunnerswitcher.h"
+
+typedef struct simpleItem_tag
+{
+    unsigned char index;
+    SINGLYLINKEDLIST_ENTRY link;
+} simpleItem, * pSimpleItem;
+
+#define TEST_CONTEXT ((const void*)0x0101)
+
+#define ENABLE_MOCKS
+
+MOCK_FUNCTION_WITH_CODE(, bool, test_match_function, PSINGLYLINKEDLIST_ENTRY, list_entry, const void*, match_context)
+MOCK_FUNCTION_END(true);
+
+MOCK_FUNCTION_WITH_CODE(, bool, test_condition_function, PSINGLYLINKEDLIST_ENTRY, list_entry, const void*, action_context, bool*, continueProcessing);
+*continueProcessing = true;
+MOCK_FUNCTION_END(true);
+
+MOCK_FUNCTION_WITH_CODE(, bool, test_condition_function_2, PSINGLYLINKEDLIST_ENTRY, list_entry, const void*, action_context, bool*, continueProcessing);
+simpleItem* item = CONTAINING_RECORD(list_entry, simpleItem, link);
+*continueProcessing = item->index != 3;
+MOCK_FUNCTION_END(true);
+
+MOCK_FUNCTION_WITH_CODE(, int, test_action_function, PSINGLYLINKEDLIST_ENTRY, list_entry, const void*, action_context, bool*, continueProcessing);
+*continueProcessing = true;
+MOCK_FUNCTION_END(0);
+
+MOCK_FUNCTION_WITH_CODE(, int, test_action_function_2, PSINGLYLINKEDLIST_ENTRY, list_entry, const void*, action_context, bool*, continueProcessing);
+simpleItem* item = CONTAINING_RECORD(list_entry, simpleItem, link);
+*continueProcessing = item->index != 3;
+MOCK_FUNCTION_END(0);
+
+MOCK_FUNCTION_WITH_CODE(, int, test_action_function_fail, PSINGLYLINKEDLIST_ENTRY, list_entry, const void*, action_context, bool*, continueProcessing);
+*continueProcessing = true;
+simpleItem* item = CONTAINING_RECORD(list_entry, simpleItem, link);
+(int)item->index;
+if (item->index == 3)
+{
+    return MU_FAILURE;
+}
+return 0;
+MOCK_FUNCTION_END();
+
+#undef ENABLE_MOCKS
+
+static TEST_MUTEX_HANDLE test_serialize_mutex;
+
+MU_DEFINE_ENUM_STRINGS(UMOCK_C_ERROR_CODE, UMOCK_C_ERROR_CODE_VALUES)
+
+static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
+{
+    ASSERT_FAIL("umock_c reported error :%" PRI_MU_ENUM "", MU_ENUM_VALUE(UMOCK_C_ERROR_CODE, error_code));
+}
+
+BEGIN_TEST_SUITE(TEST_SUITE_NAME_FROM_CMAKE)
+
+TEST_SUITE_INITIALIZE(TestClassInitialize)
+{
+    int result;
+
+    test_serialize_mutex = TEST_MUTEX_CREATE();
+    ASSERT_IS_NOT_NULL(test_serialize_mutex);
+
+    umock_c_init(on_umock_c_error);
+
+    result = umocktypes_bool_register_types();
+    ASSERT_ARE_EQUAL(int, 0, result);
```

Comment:
You can write this as a one line:
ASSERT_ARE_EQUAL(int, 0, umocktypes_bool_register_types());

Reply:
Pretty sure that if you do that for all of these here, result is not needed

Reply:
done